game_server_launcher/launcher.py

```diff
@@ -221,6 +221,9 @@ def handle_add_player_message(self, msg):
         if msg.ip:
             self.logger.info('launcher: login server added player %d with ip %s' % (msg.unique_id, msg.ip))
             modify_firewall('whitelist', 'add', msg.unique_id, msg.ip)
+            if self.game_controller:
```

Comment:
In what case is there no game_controller?

Reply:
I guess there should always be a game controller here. I added the if because of the similar one in handle_pings_message. Changing.